### PR TITLE
Bounding box

### PR DESCRIPTION
**Pull request by A-Derks**

### CODE REVIEW

File: dkist/net/tests/test_attr_walker.py [hunk 1]
```diff
@@ -53,6 +98,18 @@ def test_walker_single(all_attrs_classes, api_param_names):
     assert not set(api_param_names[all_attrs_classes]).difference(params[0].keys())
 
 
+def test_boundingbox_containing(boundingbox_param):
+    da.BoundingBox(bottom_left= boundingbox_param[0], top_right= boundingbox_param[1],
```
Comment (Cadair) [inline dkist/net/tests/test_attr_walker.py:102]: Once you have constructed these classes you need to pass them into `walker.create` and check the output of that function yields the correct arguments for the dataset search API.

File: dkist/net/tests/test_attr_walker.py [hunk 2]
```diff
@@ -20,6 +22,41 @@ def query_or_instrument():
...
+    Create possible bounding box input coordinates and args
+    for inputs to the bounding box tests.
+    """
+    bottom_left_icrs = SkyCoord([ICRS(ra=1 * u.deg, dec=2 * u.deg, distance=150000000 * u.km)],
+                                  obstime='2021-01-02T12:34:56')
+    top_right_icrs = SkyCoord([ICRS(ra=3 * u.deg, dec=4 * u.deg, distance=150000000 * u.km)],
+                                obstime='2021-01-02T12:34:56')
+    bottom_left_vector_icrs = SkyCoord([ICRS(ra=1 * u.deg, dec=2 * u.deg, distance=150000000 * u.km),
+                                   ICRS(ra=3 * u.deg, dec=4 * u.deg, distance=150000000 * u.km)],
+                                  obstime='2021-01-02T12:34:56')
```
Comment (Cadair) [inline dkist/net/tests/test_attr_walker.py:37]: small nitpick, can you clean up the indentation here so things line up with the bracket they are enclosed in?

File: CHANGELOG.md [hunk 3]
```diff
@@ -0,0 +1,506 @@
+- added bounding box tests
```
Comment (Cadair) [inline CHANGELOG.md:1]: You don't need to add this file. To add a changelog entry for this PR follow the instructions here: https://github.com/DKISTDC/dkist/tree/main/changelog

**Review by Cadair — CHANGES_REQUESTED** [review]
Generally looks good. Can you make the codestyle check pass? I will fix the other CI issue in a separate PR.

File: dkist/net/tests/test_attr_walker.py (hunk 2 above)
Comment (A-Derks) [reply dkist/net/tests/test_attr_walker.py:37]: Yep! The 'obstime' kwarg belongs to 'SkyCoord'. Hopefully this next push clears that up.

File: dkist/net/tests/test_attr_walker.py [hunk 4]
```diff
@@ -53,6 +97,44 @@ def test_walker_single(all_attrs_classes, api_param_names):
...
 
+def test_boundingbox_containing(boundingbox_param):
+    bb_query = da.BoundingBox(bottom_left= boundingbox_param[0], top_right= boundingbox_param[1],
+                       width= boundingbox_param[2], height=boundingbox_param[3], search="containing")
+
+    out = walker.create(bb_query)
+    assert len(out) == 1
+    assert all([isinstance(a, dict) for a in out])
+
+    #can't verify exact coordinates, they change a bit
```
Comment (Cadair) [inline dkist/net/tests/test_attr_walker.py:108]: we need at least one test case which checks we are formatting the coordinates correctly, as the dataset search api expects them in a given form.

File: dkist/net/tests/test_attr_walker.py [hunk 5]
```diff
@@ -53,6 +99,60 @@ def test_walker_single(all_attrs_classes, api_param_names):
...
+        assert key == 'rectangleContainingBoundingBox'
+
+    for value in out[0].values():
+        # want to make sure the value is of the format (flt, flt), (flt, flt)
+        coordinate_regex = re.compile(r'^(\()(-?\d+)(\.\d+)?(,)(-?\d+)(\.\d+)?(\))(,)(\()(-?\d+)(\.\d+)?(,)(-?\d+)(\.\d+)?(\))$')
+        coordinate_regex.search(value)
+
+
+
+
```
Comment (Cadair) [inline dkist/net/tests/test_attr_walker.py:121]:
```suggestion
```

File: dkist/net/tests/test_attr_walker.py [hunk 6]
```diff
@@ -53,6 +99,60 @@ def test_walker_single(all_attrs_classes, api_param_names):
...
+    # can't verify exact coordinates, they change a bit
+    for key in out[0].keys():
+        assert key == 'rectangleContainedByBoundingBox'
+
+    for value in out[0].values():
+        # want to make sure the value is of the format (flt, flt), (flt, flt)
+        coordinate_regex = re.compile(r'^(\()(-?\d+)(\.\d+)?(,)(-?\d+)(\.\d+)?(\))(,)(\()(-?\d+)(\.\d+)?(,)(-?\d+)(\.\d+)?(\))$')
+        coordinate_regex.search(value)
+
```
Comment (Cadair) [inline dkist/net/tests/test_attr_walker.py:138]:
```suggestion


```

File: dkist/net/tests/test_attr_walker.py [hunk 7]
```diff
@@ -53,6 +99,60 @@ def test_walker_single(all_attrs_classes, api_param_names):
     assert not set(api_param_names[all_attrs_classes]).difference(params[0].keys())
 
 
+def test_boundingbox_containing(boundingbox_param):
```
Comment (Cadair) [inline dkist/net/tests/test_attr_walker.py:102]: Could we parametrize these tests with a `(kwarg, query_param)` key? as the only things which are changing between the three tests are the input and the expected strings, seems silly to write out the same test three times for that?

If you haven't user parametrize before I am talking about this: https://docs.pytest.org/en/latest/example/parametrize.html

**Review by Cadair — CHANGES_REQUESTED** [review]
Just style changes now.

You can accept any of my suggestions in a batch by going to the "files" tab and adding each one. GitHub will then commit them to the PR, or you can apply them manually, up to you.

File: dkist/net/tests/test_attr_walker.py (hunk 7 above)
Comment (A-Derks) [reply dkist/net/tests/test_attr_walker.py:102]: Yep it does seem repetitive to type out the same test 3x. I did think about this, but the tests are already parametrized with boundingbox_param. Can you double parametrize? i.e. have one boundingbox_param and also some expected string param?

File: dkist/net/tests/test_attr_walker.py [hunk 8]
```diff
@@ -53,6 +99,30 @@ def test_walker_single(all_attrs_classes, api_param_names):
...
+@pytest.mark.parametrize("search,search_type",
+                         [
+                             ('containing', 'rectangleContainingBoundingBox'),
+                             ('contained', 'rectangleContainedByBoundingBox'),
+                             ('intersecting', 'rectangleIntersectingBoundingBox'),
+                         ]
+                         )
+def test_boundingbox(search, search_type, boundingbox_param):
+    bb_query = da.BoundingBox(bottom_left= boundingbox_param[0], top_right= boundingbox_param[1],
+                       width= boundingbox_param[2], height=boundingbox_param[3], search=search)
```
Comment (Cadair) [inline dkist/net/tests/test_attr_walker.py:111]:
```suggestion
    bb_query = da.BoundingBox(bottom_left=boundingbox_param[0], top_right=boundingbox_param[1],
                       width=boundingbox_param[2], height=boundingbox_param[3], search=search)
```

**Review by Cadair — APPROVED** [review]

**Pull request merged**